src/runtime/capabilities.ts

```diff
@@ -8,19 +8,44 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {RecipeHandleCapability} from './manifest-ast-nodes.js';
+import {AnnotationRef} from './recipe/annotation.js';
+import {Dictionary} from './hot.js';
+
+export enum Capability {
+  Persistent = 'persistent',
+  Queryable = 'queryable',
+  TiedToRuntime = 'tiedToRuntime',
+  TiedToArc = 'tiedToArc',
+}
 
 export class Capabilities {
-  private readonly capabilities: Set<RecipeHandleCapability>;
+  private readonly capabilities: Set<Capability>;
 
-  constructor(capabilities: RecipeHandleCapability[]) {
+  constructor(capabilities: Capability[]) {
     this.capabilities = new Set(capabilities);
   }
 
-  get isPersistent() { return this.capabilities.has('persistent'); }
-  get isQueryable() { return this.capabilities.has('queryable'); }
-  get isTiedToRuntime() { return this.capabilities.has('tied-to-runtime'); }
-  get isTiedToArc() { return this.capabilities.has('tied-to-arc'); }
+  static fromAnnotations(annotations: AnnotationRef[]) {
+    const capSet = new Set<Capability>();
+    for (const ann of annotations) {
```

Comment:
Nit: it's preferable to spell things out fully rather than use abbreviations or acronyms, unless they're very well understood: `capabilitySet`, `annotation`

Reply:
renamed.